lib/ledger_sync/adaptors/ledger_serializer_attribute_set.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module LedgerSync
+  module Adaptors
+    class LedgerSerializerAttributeSet
+      attr_reader :attributes,
+                  :id_attribute,
+                  :ledger_attribute_keyed_hash,
+                  :resource_attribute_keyed_hash,
+                  :serializer_class
+
+      delegate  :[],
+                :each,
+                :include?,
+                :map,
+                to: :attributes
+
+      def initialize(serializer_class:)
+        @attributes = []
+        @ledger_attribute_keyed_hash = {}
+        @resource_attribute_keyed_hash = {}
+        @serializer_class = serializer_class
+      end
+
+      def add(attribute)
+        @attributes << attribute
+        @id_attribute = attribute if attribute.id
+        if attribute.ledger_attribute.present?
+          # raise "ledger_attribute already defined for #{serializer_class.name}: #{attribute.ledger_attribute}" if ledger_attribute_keyed_hash.key?(attribute.ledger_attribute.to_s)
```

Comment:
No, it needs removed or uncommented.  I believe `ledger_attribute` is fine to enforce there is only one definition.  

I thought `resource_attribute` on the other hand is not, as you could in theory use the same value multiple places when calling `to_ledger_hash`.  BUT that also becomes an issue because only one value will survive when calling `deserialize`.

So perhaps disallowing duplicates on either is OK for now, until we have a real use case we need to support?

Reply:
Up to you. I'm fine disallowing